File: Jump_Game/jump_game.dart
class Solution {
  bool canJump(List<int> nums) {
    var maxSteps = 0;

    for (var i = 0; i < nums.length; i++) {
      if (i > maxSteps) {
        return false;
      }

      if (i + nums[i] > maxSteps) {
        maxSteps = i + nums[i];
      }
    }

    return true;
  }
}

File: Jump_Game/jump_game.go
package jump_game

func canJump(nums []int) bool {
	maxSteps := 0
	for i := 0; i < len(nums); i++ {
		if i > maxSteps {
			return false
		}

		if i+nums[i] > maxSteps {
			maxSteps = i + nums[i]
		}
	}

	return true
}
